internal/app/api/handler/login/login.go

```diff
@@ -3,39 +3,54 @@ package login
 import (
 	"net/http"
 
+	"github.com/globalsign/mgo/bson"
 	"github.com/gorilla/mux"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/bcrypt"
 
-	"github.com/TheFlies/ofriends/internal/app/service"
+	"github.com/TheFlies/ofriends/internal/app/types"
 	"github.com/TheFlies/ofriends/internal/pkg/glog"
 	"github.com/TheFlies/ofriends/internal/pkg/respond"
 )
 
 type (
 	LoginHandler struct {
-		srv      service.UserService
+		srv      UserService
 		logger   glog.Logger
-		loginSvr service.Loginservice
+		loginSvr []LoginService
+	}
+	UserService interface {
+		GetByName(username string) (*types.User, error)
+		AddUser(user *types.User) error
+		CheckExistence(username string) bool
+		ChangeUserPassword(user *types.User) error
+	}
+	LoginService interface {
+		Authenticate(username string, password string) (interface{}, error)
 	}
 )
 
-func NewLoginHandeler(s service.UserService, loginsvr service.Loginservice, l glog.Logger) *LoginHandler {
+func NewLoginHandeler(s UserService, loginsvr []LoginService, l glog.Logger) *LoginHandler {
 	return &LoginHandler{
 		srv:      s,
 		logger:   l,
 		loginSvr: loginsvr,
 	}
 }
 func (h *LoginHandler) Authenticate(w http.ResponseWriter, r *http.Request) {
-	username := r.FormValue("username")
-	password := r.FormValue("password")
-	h.logger.Infof("login with username : %v,password:****** ", username)
-	respondMap, err := h.loginSvr.Authenticate(username, password)
-	if err != nil {
-		respond.JSON(w, http.StatusInternalServerError, "login fail")
-		return
+	h.logger.Infof("Login with %d method ", len(h.loginSvr))
+	Username := r.FormValue("username")
+	Password := r.FormValue("password")
+	h.logger.Infof("login with Username : %v,Password:****** ", Username)
+	for _, loginsvr := range h.loginSvr {
+		respondMap, err := loginsvr.Authenticate(Username, Password)
+		if err == nil {
+			respond.JSON(w, http.StatusAccepted, respondMap)
+			return
+		}
 	}
-	respond.JSON(w, http.StatusAccepted, respondMap)
+	respond.JSON(w, http.StatusUnauthorized, "login fail")
```

Comment:
the response json should be something meaningful so that you can use it to check/handle logic in UI. {"status":403, "message": "failed to login"} is an example.